src/components/context/utilities.ts

```diff
@@ -19,11 +21,15 @@ import {
   TextAppDataDataRecord,
 } from '../types/types';
 
+export const generateId = (): string => {
+  return uuidv4();
+}
+
 export const getQuestionById = (
   data: List<QuestionDataAppSettingRecord>,
   id: string
 ) => {
-  return data?.filter((d) => d.id === id)?.first();
+  return data?.filter((d) => d.data.questionId === id)?.first();
```

Comment:
```suggestion
  return data.find((d) => d.data.questionId === id);
```
I've removed `?` as well, but if it crashes, then you need to change the argument of the function as optional (`data?`).